src/compiler/checker.ts

```diff
@@ -30184,8 +30184,9 @@ namespace ts {
                         workStacks.leftType[stackIndex] = leftType;
                         const operator = node.operatorToken.kind;
                         if (operator === SyntaxKind.AmpersandAmpersandToken || operator === SyntaxKind.BarBarToken || operator === SyntaxKind.QuestionQuestionToken) {
-                            if (operator === SyntaxKind.AmpersandAmpersandToken) {
-                                checkTestingKnownTruthyCallableType(node.left, leftType, isIfStatement(node.parent) ? node.parent.thenStatement : undefined);
+                            if (strictNullChecks && operator === SyntaxKind.AmpersandAmpersandToken) {

```

Comment:
doesn't `checkTestingKnownTruthyCallableType` immediately check `strictNullChecks`? I don't think you need to add it here.

Reply:
You are right, `checkTestingKnownTruthyCallableType` has `strictNullChecks` check, I've added it just to skip calling `walkUpParenthesizedExpressions` in non `strict` mode. Does that make sense?

Reply:
You are right, `checkTestingKnownTruthyCallableType` has `strictNullChecks` check, I've added it just to skip calling `walkUpParenthesizedExpressions` in non strict mode. Does that make sense?

Reply:
mmm I guess so. The downside is in case we decide to extend checking to loose mode, this will prevent that.

Never mind, I talked myself out of it. `walkUp...` is usually cheap, a single `kind` check, so I think it's OK to call it every time. It's worth it to only have one copy of the decision whether to check or not.

Reply:
@sandersn Oke, thanks. I've removed `strictNullChecks` from the condition.